src/redux/books/booksSlice.js

```diff
@@ -1,40 +1,123 @@
-import { createSlice } from '@reduxjs/toolkit';
+// import { createSlice } from '@reduxjs/toolkit';
 
+// const initialState = {
+//   books: [{
+//     item_id: 'item1',
+//     title: 'The Great Gatsby',
+//     author: 'John Smith',
+//     category: 'Fiction',
+//   },
+//   {
+//     item_id: 'item2',
+//     title: 'Anna Karenina',
+//     author: 'Leo Tolstoy',
+//     category: 'Fiction',
+//   },
+//   {
+//     item_id: 'item3',
+//     title: 'The Selfish Gene',
+//     author: 'Richard Dawkins',
+//     category: 'Nonfiction',
+//   }],
+// };
+
+// const booksSlice = createSlice({
+//   name: 'books',
+//   initialState,
+//   reducers: {
+//     addBook: (state, action) => ({
+//       ...state,
+//       books: [...state.books, action.payload],
+//     }),
+//     removeBook: (state, action) => ({
+//       ...state,
+//       books: state.books.filter((book) => book.itemID !== action.payload),
+//     }),
+//   },
+// });
+
+// export const { addBook, removeBook } = booksSlice.actions;
+// export default booksSlice.reducer;
+
+/// ////////////////////////////////////////////////////////////////////////////////////////////////
+
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+// Initial State
 const initialState = {
-  books: [{
-    item_id: 'item1',
-    title: 'The Great Gatsby',
-    author: 'John Smith',
-    category: 'Fiction',
+  books: [],
+  isLoading: false,
+  error: null,
+};
+
+// Async Thunks
+export const fetchBooksList = createAsyncThunk(
+  'books/fetchBooksList',
+  async () => {
+    const response = await axios.get('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/d5U1jci9rU4Ou9MD2rVE/books');
+    const { data } = response;
+    const books = Object.keys(data).map((id) => ({
+      id,
+      ...data[id][0],
+    }));
+
+    return books;
+  },
+);
+
+export const addBook = createAsyncThunk(
+  'books/addBook',
+  async (book) => {
+    const response = await axios.post('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/d5U1jci9rU4Ou9MD2rVE/books', book);
+    // return response.data;
```

Comment:
- [ Required ] Please consider removing any commented pieces of code from your project, commented code makes your project look unprofessional and is not considered clean code.